src/runtime.rs

```diff
@@ -5,17 +5,43 @@
 // accordance with the terms of the Adobe license agreement accompanying
 // it.
 
-use std::sync::OnceLock;
+use crate::settings::get_global_settings_toml;
+use c2pa::settings::Settings;
+use std::sync::{Arc, OnceLock, RwLock};
 use tokio::runtime::{Builder, Runtime};
 
-// Store tokio runtime in a singleton
-static RUNTIME: OnceLock<Runtime> = OnceLock::new();
+// Runtime stored in a swap-able singleton to allow reloads after settings change
+static RUNTIME: OnceLock<RwLock<Arc<Runtime>>> = OnceLock::new();
 
-pub fn runtime() -> &'static Runtime {
-    RUNTIME.get_or_init(|| {
-        Builder::new_multi_thread()
-            .enable_all()
-            .build()
-            .expect("Failed to build runtime")
-    })
+fn build_runtime() -> Arc<Runtime> {
+    let rt = Builder::new_multi_thread()
+        .enable_all()
+        .on_thread_start(|| {
+            // Apply the latest global TOML snapshot when each worker starts
```

Comment:
When people set JSON settings... Are those applied on the fly then? THe TOML ones are the defaults?